test/staticvector_test.cpp

```diff
@@ -0,0 +1,82 @@
+#include <cstdlib>
+#include <ctime>
+#include <gtest/gtest.h>
+
+#include "utils/static_vector.hpp"
+
+#define MAX_SIZE 32
+
+using namespace devilution;
+
+namespace {
+StaticVector<int, MAX_SIZE> container;
+
+TEST(StaticVector, StaticVector_push_back)
+{
+	std::srand(clock());
+	size_t size = 10 + (std::rand() % (MAX_SIZE - 10));
+	for (size_t i = 0; i < size; i++) {
+		container.push_back(i);
+	}
+
+	for (size_t i = 0; i < container.size(); i++) {
+		EXPECT_EQ(container[i], i);
+	}
+}
+
+TEST(StaticVector, StaticVector_erase)
+{
+	std::srand(clock());
+
+	if (container.size() == 0) {
+		return;
+	}
+
+	int erasures = std::rand() % container.size();
+
+	std::vector<int> erase_idx;
+	std::vector<int> expected;
+
+	erase_idx.push_back(0);
+	erase_idx.push_back(container.size() - 1);
+
+	for (int i = 0; i < erasures; i++) {
+		erase_idx.push_back(std::rand() % container.size());
+	}
+
+	for (int i = 0; expected.size() < container.size(); i++) {
+		expected.push_back(i);
+	}
+
+	for (int idx : erase_idx) {
+		/* This is actually to guard against gcc's vector implementation causing
+		 * crashes, rather than StaticVector
+		 */
+		if (expected.size() == 0) {
+			break;
+		}
+		container.erase(container.begin() + idx, container.begin() + idx + 1);
+		expected.erase(expected.begin() + idx);
+	}

```

Comment:
Deleting an element from these containers changes the size of the containers, meaning all indexes greater than the deleted location no longer point to the originally intended location. If you want to handle random deletions you'll need to ensure you either delete from the end first (e.g. reverse sort the indexes to delete) or adjust your indexes to ensure they point to valid locations (e.g. decrement every index > the one you just deleted.